plugins/GrapesJsBuilderBundle/EventSubscriber/SerializerSubscriber.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace MauticPlugin\GrapesJsBuilderBundle\EventSubscriber;
+
+use JMS\Serializer\EventDispatcher\Events;
+use JMS\Serializer\EventDispatcher\EventSubscriberInterface;
+use JMS\Serializer\EventDispatcher\ObjectEvent;
+use JMS\Serializer\JsonSerializationVisitor;
+use JMS\Serializer\Metadata\StaticPropertyMetadata;
+use Mautic\EmailBundle\Entity\Email;
+use Mautic\InstallBundle\Install\InstallService;
+use MauticPlugin\GrapesJsBuilderBundle\Integration\Config;
+use MauticPlugin\GrapesJsBuilderBundle\Model\GrapesJsBuilderModel;
+
+class SerializerSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private GrapesJsBuilderModel $grapesJsBuilderModel,
+        private InstallService $installer,
+        private Config $config,
+    ) {
+    }
+
+    /**
+     * @return array<int, array<string, string>>
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            [
+                'event'  => Events::POST_SERIALIZE,
+                'method' => 'addCustomMJML',
+            ],
+        ];
+    }
+
+    public function addCustomMJML(ObjectEvent $event): void
+    {
+        if (!$this->installer->checkIfInstalled()) {
```

Comment:
Please order the checks by how expensive they are. For example checking an instance of an object is way cheeper than checking if Mautic is installed as that is accessing filesystem. Do we even need this check? If Mautic isn't installed, how did you generate credentials?

Reply:
Yes I agree that we won't need the Mautic check.